tasks/banana.js

```diff
@@ -156,6 +158,20 @@ module.exports = function ( grunt ) {
 				} );
 			}
 
+			if ( options.requireKeyPrefix.length ) {
+				if ( typeof options.requireKeyPrefix === 'string' ) {
+					options.requireKeyPrefix = [ options.requireKeyPrefix ];
+				}
+				sourceMessageWrongPrefix = sourceMessageKeys.filter( function ( key ) {
+					return options.requireKeyPrefix
+						.map( function ( prefix ) {
+							return !key.startsWith( prefix );
+						} ).reduce( function ( failed, accumulator ) {
```

Comment:
Today I learned more about `reduce()` than I thought there was.

* This function is not at all run if only one prefix is provided. Instead, it is internally resolved to the first value returned by `map()` as-is.
* This function starts its iteration at the second value (index 1 and above), when only a callback is given (no initial value for the accumulator), where accumulator will be the first value.
* Only if the second parameter `initialValue` is set for `reduce()`, will it iterate over each value.

I thought it was a mistake and presumed `accumulator` would be undefined and this always combine into a false-y value.

Anyway, this is pretty cool, and I think this makes reduce more useful than I previously thought. It's not a hack.

But.. I do think in this particular case it would be cleaner and simpler to avoid it. In favour of a single filter operation. Or even `some()`.

```js
 // Current
 return prefixes.map( function ( prefix ) {
   return !key.startsWith( prefix ); // list of bools, true if the key doesn't use the prefix
 } ).reduce( function ( failed, accumulator ) {
   return accumulator && failed; // assert they are all true ("failed", unused prefixes)
 } ); // true if the message key used none of the prefixes

 // Proposed
 return prefixes.filter( function ( prefix ) {
   return key.startsWith( prefix ); // find which prefix(es) match the message
 } ).length === 0; // assert the message matched none of the prefixes

 // Proposed (alt)
 return !prefixes.some( function ( prefix ) { // assert the message doesn't match even one prefix
   return key.startsWith( prefix );
 } );
```